### ISSUE TITLE
non-function casting operator do not work with `in array`

### ISSUE THREAD

**Issue by leonyu** [decaffeinate/decaffeinate#499]
decaffeinate is crashing on my CoffeeScript input:

``` coffee

x = "#{abc}" in ['a', 'b']
y = not bool in [true, false]

```

http://decaffeinate-project.org/repl/#?evaluate=true&stage=full&code=y%20%3D%20%22%23%7Babc%7D%22%20in%20%5B'a'%2C%20'b'%5D%0Ax%20%3D%20not%20bool%20in%20%5Btrue%2C%20false%5D%0A

I get this error:

```
AddVariableDeclarationsStage failed to parse: Unexpected token, expected , (1:42)
> 1 | x = (ref = `${abc}` === 'a' || ref === 'b';
    |                                           ^
  2 | y = !bool === true || not bool === false;
  3 | 
```

**Comment by alangpierce** [comment]
Thanks for reporting. I think this will be fixed with the fix for #482.

**Comment by alangpierce** [comment]
Ok, this was fixed with #508.
